### PR TITLE
Interface Task Commit(Maria)

### PR DESCRIPTION
**Pull request by mariastephan2186**
Electronic Devices Interfaces task done

### CODE REVIEW

File: src/main/java/session10overridingandoverloading/examples/interfaces/InterfacesTask_Maria/Refrigerator.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    @Override
+
+    public void turnOff() {
+        System.out.println("Refrigerator is turned off");
+    }
+
+    @Override
+
+    public boolean isOn() {
+        return false;
```
Comment (owynowyn) [inline src/main/java/session10overridingandoverloading/examples/interfaces/InterfacesTask_Maria/Refrigerator.java:19]: State Management:
Currently, the isOn method always returns false, indicating that the lamp is always off. You should maintain a state variable to track whether the lamp is on or off.

`private boolean isOn = false;` as a field at the top of each class to begin with, then in your methods you could set isOn to true/false accordingly.

File: src/main/java/session10overridingandoverloading/examples/interfaces/InterfacesTask_Maria/Refrigerator.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+public class Refrigerator implements ElectronicDevice {
+
+    @Override
+    public void turnOn() {
+        System.out.println("Refrigerator is turned on , food is cold ");
+    }
+
+    @Override
+
+    public void turnOff() {
```
Comment (owynowyn) [inline src/main/java/session10overridingandoverloading/examples/interfaces/InterfacesTask_Maria/Refrigerator.java:12]: Logical Implementation:

The turnOn and turnOff methods should check the current state before performing an action.
The isOn method should return the actual state of the refrigerator.

```
private boolean on = false; // state variable to track if the refrigerator is on or off

    @Override
    public void turnOn() {
        if (!on) { // if the refrigerator is off, turn it on
            on = true;
            System.out.println("Refrigerator is turned on, food is cold.");
        } else {
            System.out.println("Refrigerator is already on.");
        }
    }
```

File: src/main/java/session10overridingandoverloading/examples/interfaces/InterfacesTask_Maria/Refrigerator.java [hunk 3]
```diff
@@ -0,0 +1,21 @@
+package session10overridingandoverloading.examples.interfaces.InterfacesTask_Maria;
+
+public class Refrigerator implements ElectronicDevice {
+
+    @Override
+    public void turnOn() {
+        System.out.println("Refrigerator is turned on , food is cold ");
```
Comment (owynowyn) [inline src/main/java/session10overridingandoverloading/examples/interfaces/InterfacesTask_Maria/Refrigerator.java:7]: Logical Implementation:

The turnOn and turnOff methods should check the current state before performing an action.
The isOn method should return the actual state of the refrigerator.

```
private boolean on = false; // state variable to track if the refrigerator is on or off

    @Override
    public void turnOn() {
        if (!on) { // if the refrigerator is off, turn it on
            on = true;
            System.out.println("Refrigerator is turned on, food is cold.");
        } else {
            System.out.println("Refrigerator is already on.");
        }
    }
```

**Comment by mariastephan2186** [comment]
@owynowyn Changes have been made as per your comments

**Review by owynowyn — APPROVED** [review]
LGTM! ✅

**Pull request merged**